File: app/src/main/java/hm/orz/chaos114/android/slideviewer/ui/CrashReportingTree.java
package hm.orz.chaos114.android.slideviewer.ui;

import android.text.TextUtils;
import android.util.Log;

import com.crashlytics.android.Crashlytics;

import timber.log.Timber;

public class CrashReportingTree extends Timber.Tree {
    @Override
    protected void log(int priority, String tag, String message, Throwable t) {
        if (priority == Log.VERBOSE || priority == Log.DEBUG) {
            return;
        }
        if (priority != Log.ERROR) {
            Crashlytics.log(message);
            return;
        }
        Throwable throwable = t;
        if (throwable == null && !TextUtils.isEmpty(message)) {
            throwable = new Exception(message);
        }
        Crashlytics.logException(throwable);
    }
}

File: app/src/main/java/hm/orz/chaos114/android/slideviewer/ui/CrashReportingTree.kt
package hm.orz.chaos114.android.slideviewer.ui

import android.text.TextUtils
import android.util.Log
import com.crashlytics.android.Crashlytics

import timber.log.Timber

class CrashReportingTree : Timber.Tree() {
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (priority == Log.VERBOSE || priority == Log.DEBUG) {
            return
        }
        if (priority != Log.ERROR) {
            Crashlytics.log(message)
            return
        }

        val throwable = if (t == null && !TextUtils.isEmpty(message)) {
            Exception(message)
        } else {
            t
        }
        Crashlytics.logException(throwable)
    }
}
